### PR TITLE
python 3.6.0

### PR DESCRIPTION
**Pull request by skycaptain**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by tdsmith** [comment]
Thanks for this! I'm inclined to bump the devel spec to 3.6.0 but leave stable on the 3.5 branch until there are numpy, scipy, and cryptography wheels for 3.6, since that seemed to be the pain point when we transitioned from 3.4 to 3.5, but I'm open to feedback.

**Comment by ilovezfs** [comment]
Nope, we should ship 3.6.0 as stable. Upstream has designated as such, and that's what people will want for the bottles.

**Comment by tdsmith** [comment]
Yes, we should and we will; the question in my mind, which Mike raised after the 3.5 transition, is whether it's helpful to do that immediately, before the packaging ecosystem has caught up.

**Comment by ilovezfs** [comment]
I don't see this as remotely controversial. If 3.6 isn't break anything in Homebrew: :ship:

**Comment by tdsmith** [comment]
With respect to the instant PR, all of the failing packages will need to have a revision bump.

**Comment by ilovezfs** [comment]
squashed, and revision bumps added.

**Comment by pirate** [comment]
Pardon me for asking if this isn't the right place, but why the revision bumps on other packages?  Is it so they will upgrade and reinstall the python3 dependency in the process in order to get python3.6?

**Comment by ilovezfs** [comment]
@pirate it forces those formulae to be rebuilt from source even though their own versions aren't changing, and then triggers upgrades for users to those newly built bottles.

**Comment by alex** [comment]
As someone who makes wheels; I'm +1 on homebrew upgrading and breaking us; without users complaining to us there's a real chance we'll forget.

**Comment by MikeMcQuaid** [comment]
I'm personally fine shipping this as-is but if @tdsmith would rather wait until other Python stuff is sorted I'm fine with that too. I definitely think if we're going to hold off on those packages being ready that they should be moved into Homebrew/homebrew-core.

**Comment by skycaptain** [comment]
Remaining errors in checks are imho innocuous and are not related to this pr.

File: Formula/python3.rb [hunk 1]
```diff
@@ -47,17 +29,17 @@ class Python3 < Formula
   depends_on :x11 if build.with?("tcl-tk") && Tab.for_name("homebrew/dupes/tcl-tk").with?("x11")
   depends_on "sphinx-doc" => [:build, :optional]
 
-  skip_clean "bin/pip3", "bin/pip-3.4", "bin/pip-3.5"
-  skip_clean "bin/easy_install3", "bin/easy_install-3.4", "bin/easy_install-3.5"
+  skip_clean "bin/pip3", "bin/pip-3.4", "bin/pip-3.5", "bin/pip-3.6"
+  skip_clean "bin/easy_install3", "bin/easy_install-3.4", "bin/easy_install-3.5", "bin/easy_install-3.6"
 
   resource "setuptools" do
```
Comment (zmwangx) [inline Formula/python3.rb]: I realize this won't be a popular opinion for people waiting for this PR to land, but I think setuptools should be upgraded to 32.2.0, or at least 32.1.0 to match the version in [`python.rb`](https://github.com/Homebrew/homebrew-core/blob/7330a126bf338b5216dc3b31bb65485cbedbbbf1/Formula/python.rb#L44-L47).

Comment (ilovezfs) [inline Formula/python3.rb]: done.

**Comment by ilovezfs** [comment]
@skycaptain thanks for your speedy pull request and your contribution to Homebrew! :snake:

**Pull request closed without merging**